### DISCUSSION TITLE
Cherry picking commits in the mobile app

### DISCUSSION THREAD

**Discussion by eliehassan** [community/community#54447] [Mobile]
<details>
<summary>ℹ️ There currently seems to be no way to Cherry Pick a commit to an new branch in the mobile app (This is available on the desktop app.</summary>
GitHub for iOS, Version 1.111.1 (98645245)
Device: iPad Pro 12.9-inch 5 (iOS 16.4.1)
</details>

**Comment by mattruggio** [comment]
Hey @eliehassan, thanks for your suggestion!  We are constantly trying to bring parity between our web, desktop, and mobile applications and this would seem to fall into that category.  I will get this added to our internal list.

**Reply by eliehass** [reply]
It's been a while so I'm just checking in to see if there is any update here. Thanks!

**Reply by mattruggio** [reply]
Hey @eliehass! I just checked, it is still in our internal backlog. I think a blocker may exist here: Desktop looks to perform a local git cherry-pick operation. This is inherently different than how mobile works: Desktop has local native git operation functionality while mobile has to rely 100% on our remote APIs for mutability. After looking through our API's, I cannot find any GraphQL mutations or REST endpoints which support remote cherry-picking capabilities. Digging a bit deeper, I am not sure I see any functionality exposed on the website which allows cherry-picking either. Usually (but not always) we will want to implement this level of functionality on the website before bringing it to our client applications (Desktop, CLI, mobile). I most certainly could have missing something here, is there potentially something I am missing on how this works on the web or APIs that you are aware of? If so, could you provide any examples?

**Reply by eliehass** [reply]
You are correct, it is not available on the website either and it would be very helpful if it was. My main issue is that sometimes I need to cherry pick a commit one of my devs have made into a different branch when I am not on my desktop and being able to do that from the app/web interface would make things a lot easier.

**Reply by torressam333** [reply]
Is there any plan to add a cherry pick button or feature to the website?

**Comment by EwoutH** [comment]
Yes, I would love being able to cherry pick a commit just from the mobile app!

> Digging a bit deeper, I am not sure I see any functionality exposed on the website which allows cherry-picking either. 

I would also love to do this on the website. Maybe together with:
- Undoing an commit
- Amending a commit
- Reverting a commit
- Rebasing a branch on another branch

If doing this from the API without a local checkout, this would be amazing!
